http-client/Network/HTTP/Client.hs

```diff
@@ -322,9 +323,14 @@ managerSetProxy :: ProxyOverride -> ManagerSettings -> ManagerSettings
 managerSetProxy po = managerSetInsecureProxy po . managerSetSecureProxy po
 
 -- @since 0.7.17
-managerSetMaxHeaderLength :: Int -> ManagerSettings -> ManagerSettings
+managerSetMaxHeaderLength :: Word -> ManagerSettings -> ManagerSettings
```

Comment:
This is a breaking change, I do not want to include it. It's not worth the ecosystem churn for a small improvement in type correctness.

Reply:
You are right, I actually thought about it, and I didn't change it.

Reply:
But... are you talking about `managerSetMaxHeaderLength` only? I would just revert to use `Word` for both `managerSetMaxHeaderLength` and `managerSetMaxNumberHeaders`, and also the newtypes used internally for distinguishing those values.

Reply:
Ah, I missed your main comment above. Personally I would avoid using `Int` in one place and `Word` in an other, it would be confusing to me: why one max value is an int an an other a word?
If we want to _refine_ a bit the types of the interface, let's do that in a new version like _0.8.0_, we could put there multiple breaking changes.

Reply:
I'm not in favor of a breaking change for this topic. I'll leave the choice to you: inconsistency in the API, or using `Int` in both places. My only stance is avoiding breaking changes.